Assets/Scripts/DestroyProjectile.cs

```diff
@@ -35,8 +35,17 @@ private void OnCollisionEnter2D(Collision2D collision)
     {
         if (collision.gameObject.tag == "Enemy")
         {
-            collision.gameObject.GetComponent<Skeleton>().DamageSkeleton(Player.instance.playerDamage);
-            gameObject.SetActive(false);
+            if (collision.gameObject.name == "Skeleton")
+            {
+                collision.gameObject.GetComponent<Skeleton>().DamageSkeleton(Player.instance.playerDamage);
+                gameObject.SetActive(false);
+            }
+            //this is if instead of else as I plan on making at least one more enemy. Not sure if I should make else until I make the other enemies or if this is fine.
```

Comment:
>//this is if instead of else as I plan on making at least one more enemy. Not sure if I should make else until I make the other enemies or if this is fine. [](start = 12, length = 155)

I'd recommend always continuing your state machine switches with an 'else if' and if you end up with some generic response that all the rest of the non special enemies do you would put that in an else at the end.  This is because while this code is running, if the object was a skeleton, it can skip checking to see if the object is a goblin since that is in an else if block.  You'll see minor performance increases doing this all over the place.

Reply:
Okay, so unless I want there to be multiple correct answers, I should always do else if statements. Are there situations where an else if statement is better than a switch statement, or should I go back and rewrite it as a switch statement?

Reply:
Switch statements are simply else if blocks, but most people think they look cleaner, up to you if you want to do that.

Since you are testing objects by name you wont run into multiple correct answers since the object can't have two names.  If you were checking based on a different property, like you had a monster type property which could be both goblin and skeleton you might not want to do and else if system.

---
In reply to: [257056689](https://github.com/SpencerAherne/CorvoAdventure/pull/5#discussion_r257056689) [](ancestors = 257056689)

Reply:
Okay so it's mostly just a matter of what I like more, good to know. And yeah in this instance I would only have one right answer, but I was curious for future additions as I do plan on making some area of effect attacks, that could potentially hit multiple different enemies objects. But I guess that still might work differently than I expect.